reanalysis/moi_tests.py

```diff
@@ -175,41 +186,119 @@ def __init__(
         :return:
         """
-        if self.pedigree.get(sample_id).affected:
-            return True
-        return False
-
-    # not implemented for singleton analysis
-    # def get_affected_parents(self, sample_id: str) -> Set[str]:
-    #     """
-    #     for this sample, find any affected parents
-    #     port this back into the Participant object?!
-    #     prevent re-computation... but really only make that more complex if necessary
-    #     :param sample_id:
-    #     :return:
-    #     """
-    #     participant = self.pedigree.get(sample_id)
-    #
-    #     return {
-    #         parent.details.sample_id
-    #         for parent in [participant.mother, participant.father]
-    #         if parent is not None and parent.details.affected
-    #     }
+
+        # initial value
+        variant_passing: bool = True
+
+        # iterate through all family members, no interested in directionality
+        # of relationships at the moment
+        for member in self.pedigree.families[self.pedigree[sample_id].family_id]:
+
+            # complete & incomplete penetrance - affected samples must have the variant
+            # complete pen. requires participants to be affected if they have the var
+            # if any of these combinations occur, fail the family
+            if (member.affected and member.sample_id not in called_variants) or (
+                member.sample_id in called_variants
+                and complete_penetrance
+                and not member.affected
+            ):
+                # fail
+                return False
+
+        return variant_passing
+
+    def check_familial_comp_het(
+        self,
+        sample_id: str,
+        called_variants_1: Set[str],
+        called_variants_2: Set[str],
+        complete_penetrance: bool = True,
+    ) -> bool:
+        """
+        compound_het check, requires 2 pools of variant calls
+        called_variants_1 & called_variants_2 are the relevant variant calls from
+        2 different variants. We evaluate the sample as 'having this variant pair'
+        if the sample ID appears in both call groups
+
+        - find the family ID from this sample
+        - iterate through all family members, and check that MOI holds for all
+        - this MOI test requires the participant to have both variant calls in order
+          to 'have this variant'
+
+        At each stage, append the sample ID of all checked samples so we don't repeat
+        One broken check will fail the variant for the whole family
+
+        Return False if a participant fails tests, True if all members pass
+
+        :param sample_id:
+        :param called_variants_1: called samples for variant 1
+        :param called_variants_2: called samples for variant 2
+        :param complete_penetrance: if True, (force affected==has variant call)
+        :return:
+        """
+
+        # initial value
+        variant_passing: bool = True
```

Comment:
This never gets updated, so maybe just return `True` in the end?

Reply:
Good spot, removed in latest commit 👍